### PR TITLE
Fixed Issue #1467 - When searching, hidden files are shown

### LINKED ISSUE #1467: When searching, hidden files are shown
Hidden files are shown when searching, even if the settings option is disabled.

### PR DESCRIPTION
**Pull request by VishnuSanal**
#### Issue tracker   
Fixes will automatically close the related issue

Fixes #1467

#### Release  
Addresses release/3.6
  
#### Test cases
- [ ] Covered
  
#### Manual testing
- [] Done  
  
If yes,  
- Device:
- OS:

#### Build tasks success  
Successfully running following tasks on local 
- [ ] `./gradlew assembledebug`
- [ ] `./gradlew spotlessCheck`

#### Related PR  
Related to PR #

#### Additional Info

Hey @EmmanuelMess, @VishalNehra,

Fixed Issue #1467 - When searching, hidden files are shown

Please let me know whether this PR meets your expectations. Please let me know if I had to do something better...

Thank You

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/SearchAsyncTask.java [hunk 1]
```diff
@@ -123,7 +124,7 @@ private void search(HybridFile directory, final SearchFilter filter) {
           activity.get(),
           rootMode,
           file -> {
-            if (!isCancelled()) {
+            if (!isCancelled() && !(file.getName().startsWith("."))) {
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/SearchAsyncTask.java:127]: Isnt there a HybridFile function to check if file is hidden?

**Comment by EmmanuelMess** [comment]
You forgot to run gradle spotlessApply.

File: app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/SearchAsyncTask.java (hunk 1 above)
Comment (VishnuSanal) [reply app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/SearchAsyncTask.java:127]: Oh! Yes. I didn't notice that. Shall do it that way?

**Comment by VishnuSanal** [comment]
> You forgot to run gradle spotlessApply.

How to `run gradle spotlessApply`.  I'm a beginner. Sorry for asking this silly question.

**Comment by EmmanuelMess** [comment]
> > You forgot to run gradle spotlessApply.
> 
> How to `run gradle spotlessApply`. I'm a beginner. Sorry for asking this silly question.

Are you on Android Studio?

**Comment by VishnuSanal** [comment]
> > > You forgot to run gradle spotlessApply.
> > 
> > 
> > How to `run gradle spotlessApply`. I'm a beginner. Sorry for asking this silly question.
> 
> Are you on Android Studio?

Yes

**Comment by EmmanuelMess** [comment]
> > > > You forgot to run gradle spotlessApply.
> > > 
> > > 
> > > How to `run gradle spotlessApply`. I'm a beginner. Sorry for asking this silly question.
> > 
> > 
> > Are you on Android Studio?
> 
> Yes

Its shift shift "gradle :spotlessApply" AFAIK

**Comment by EmmanuelMess** [comment]
@VishalNehra should this also remove the files from regex search?

**Comment by VishalNehra** [comment]
@EmmanuelMess yes

**Comment by VishalNehra** [comment]
btw shouldn't this actually show hidden files if the preference is selected in settings? current change will always filter hidden files

**Comment by VishnuSanal** [comment]
Done... Please check...

File: app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/SearchAsyncTask.java [hunk 2]
```diff
@@ -123,7 +126,11 @@ private void search(HybridFile directory, final SearchFilter filter) {
           activity.get(),
           rootMode,
           file -> {
-            if (!isCancelled()) {
+            boolean showHiddenFiles =
+                PreferenceManager.getDefaultSharedPreferences(activity.get())
+                    .getBoolean(PREFERENCE_SHOW_HIDDENFILES, false);
+
+            if (!isCancelled() && (showHiddenFiles || !(!showHiddenFiles && file.isHidden()))) {
```
Comment (VishalNehra) [inline app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/SearchAsyncTask.java:133]: Condition can be improved
`(!isCancelled() && (showHiddenFiles || !file.isHidden()))`

**Review by VishalNehra — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/SearchAsyncTask.java (hunk 2 above)
Comment (VishnuSanal) [reply app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/SearchAsyncTask.java]: > Condition can be improved
> `(!isCancelled() && (showHiddenFiles || !file.isHidden()))`

✅ Done.. Please check...

**Review by VishalNehra — APPROVED** [review]

**Review by EmmanuelMess — APPROVED** [review]

**Comment by EmmanuelMess** [comment]
@VishalNehra can 3.6 PRs be merged?

**Comment by VishalNehra** [comment]
Waiting for 3.5.2 release, so that we don't spend too much time fixing conflicts.

**Pull request merged**